### ISSUE TITLE
Proposal to show relative indent in error message

### ISSUE THREAD

**Issue by MyDogTom** [shyiko/ktlint#240]
```kotlin
class Foo {
    fun Bar(): String {
         return "bar"
    }
}
```
In current version: 
>Unexpected indentation (9) (it should be 8)

I propose, to tell about indent relative to the parent.
>Unexpected indentation (5) (it should be 4)

Reasons:
* Style guides always tell about relative indent and never about absolute.
* Current implementation of the rule knows, only correct relative indent. Correct absolute indent is unknown. Proposal allows to avoid confusion when parent block is misaligned and child block is misaligned.
* When parent block is misaligned but child block is aligned properly (relatively to parent), we don't show additional error for child block. Because entire parent block should be fixed.  This confronts with current error message, because absolute indent for child block is wrong, while relative indent is correct.

I think that such error message is more consistent with current Rule implementation. When we change Rule implementation and start calculate absolute indent, we will change error message accordingly.

Also, probably it makes sense to rephrase error message a bit, to make it more obvious that we are talking about relative indent. I have several variants:
1. "Unexpected indentation (5) (it should be 4)"
2. "Unexpected relative indentation (5) (it should be 4)"
3. "Unexpected indentation (5) relative to parent (it should be 4)"

Which one do you prefer? Do you have other ideas?

PS. I'm happy to do the change once we agree on it.

**Comment by shyiko** [comment]
> Style guides always tell about relative indent and never about absolute.

I'm not sure I've ever seen a tool that would report indents relative to the previous block. I'm not saying it's wrong. It's just, personally, whenever I think about indentantion it's mostly in the context of "indentation level", which an absolute value. I asked a few people in the office (not a representative sample, I agree) and all of them said exactly the same thing. In other words, showing relative +1, -5 indent can be confusing. We can add a CLI option to enable such output, though (if this is something people want). 

> Current implementation of the rule knows, only correct relative indent. Correct absolute indent is unknown. Proposal allows to avoid confusion when parent block is misaligned and child block is misaligned.

I don't think we should base our decision on what is obviously an incomplete/temporary implementation. I hope this doesn't sound harsh. This is not my intent. All I mean is - current implementation of IndentRule is not doing much on purpose (a year and a half ago there were no official recommendations, continuous indent was used left and right. I wanted to avoid as much contention as possible).

> When parent block is misaligned but child block is aligned properly (relatively to parent), we don't show additional error for child block.

This is an interesting point. But. I wonder what would be the output for 
```kotlin
1: fun a() {
2:         fun b() {
3:         fun c() {
4:
5:         }    
6:     }
7: }
```
`c` is misaligned relative to `b`, `b` relative to `a` and yet the only line which should be flagged is 2.
NOTE: lint errors can be grouped/collapsed over the "line range" regardless of "relative"/"absolute" approach.

> When we change Rule implementation and start calculate absolute indent, we will change error message accordingly.

I think it would be best if we rewrite IndentRule to calculate indentation in a  single-pass like we discussed before. I see no point in making 2 changes instead of 1.
